### ISSUE TITLE
[Rule request] Single spaces

### ISSUE THREAD

**Issue by rhys-vdw** [palantir/tslint#3787]
I would like a rule to enforce the following:

```ts
// BAD!
const x     = 5
const y = x === 5     ? "foo" : "bar"

// good!
const x = 5
const y = x === 5 ? "foo" : "bar"
```

It seems `whitespace` will add a minimum spacing for many operators, but not limit a maximum number of spaces.

**Comment by ggarek** [comment]
i guess, we are speaking about the rule as [`eslint/no-multi-spaces`](https://eslint.org/docs/rules/no-multi-spaces)

**Comment by timocov** [comment]
Also tslint (and seems that eslint too) allows using tabs as separator:

```ts
const abc	=	123; // actually here are tabs instead of spaces around `=`
```

I guess it will be useful to enforce to use spaces (or tabs? 🙂).

**Comment by ggarek** [comment]
imo, it is better be an option, rather than a restriction. Because using `tabs` is completely valid and common, as well as using `spaces`.

**Comment by ceirbus** [comment]
This would be so nice, my company is looking for a way to block multiple spaces before and after the equals operator (like this: '  =  ') for variable declarations. I think this could be done by borrowing from the rule for typedef-whitespace but I haven't looked into it extensively.

**Comment by JoshuaKGoldberg** [comment]
Since this is already in `tslint-eslint-rules` and the OP has otherwise solved the issue, closing this thread for housekeeping purposes. Thanks!

Relevant discussion around where these kinds of formatting rules should live: #628
